src/Compilers/CSharp/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.cs

```diff
@@ -19,6 +19,7 @@ internal sealed class DynamicAnalysisInjector : CompoundInstrumenter
         private readonly MethodSymbol _method;
         private readonly BoundStatement _methodBody;
         private readonly MethodSymbol _createPayload;
+        private readonly PooledHashSet<DebugSourceDocument> _documentsBuilder;
```

Comment:
> The only difference is where the check is done, in AddAnalysisPoint or in CreateBlockPrologue.

There is the additional difference that when its done in AddAnalysisPoint, we avoid one extra iteration over all the spans in the method to figure out the set of documents that these spans are part of.

However, I think what you are saying is that the extra iteration by itself is not expected to add significant cost. I will go ahead and change this as you propose.